backend-lambda/services/productService.js

```diff
@@ -19,25 +19,54 @@ const getAllProducts = async (pool, limit, offset) => {
 
 // NOTE: This function is untested against cockroach
 const updateProduct = async (pool, id, product) => {
-  let sets = [];
-  for (let key in product) {
-    sets.push(Format("%I = %L", key, product[key]));
-  }
+  poolCheck(pool);
+  productCheck(product);
+
+  const values = Object.values(product);
+  const keys = Object.keys(product);
 
-  let setStrings = sets.join(",");
+  // Creates keys and values array then maps them to one long-ass string ->
+  // ex: 'name = 'purple jumpsuit', size = 'Medium', description = 'comes with pockets', sold_date = '6.29.2024''
+  const setString = keys
+    .map((key, index) => Format("%I = %L", key, values[index]))
+    .join(", ");
 
-  const query = "UPDATE product SET %s WHERE product_id = %L";
-  const params = [setStrings, id];
+  /* 
+  The setString does not require a placeholder in the query nor to be in the parameters variable.
+  This is because the pg library formats the query so that the long-ass string is put into it before
+  it is passed to clientService ->
+  
+  ex:
+  UPDATE product SET name = 'purple jumpsuit', size = 'Medium', description = 'comes with pockets', sold_date = '6.29.2024'
+  WHERE product_id = $1
 
-  return clientService(pool, query, params);
+  The product_id value is the only placeholder used here with postgres's $1 placeholder syntax
+  */
+  const query = `UPDATE product SET ${setString} WHERE product_id = $1`;
+  const params = [id];
+
+  const res = await clientService(pool, query, params);
+  return res?.rows;
 };
 
 // NOTE: This function is untested against cockroach
 const createProduct = async (pool, product) => {
-  const query = "INSERT INTO product (%I) VALUES (%L)";
-  const params = [Object.keys(product), Object.values(product)];
+  const keys = Object.keys(product).join(", ");
```

Comment:
Aha now this method has SQL injection possibilities all over it. Let me know if you want help working through it. https://portswigger.net/web-security/sql-injection

Reply:
So that values look they get sanitized, but the keys do not.

Reply:
So it's just the keys that are vulnerable.

Reply:
So the query will use placeholder values - won't that prevent sql injection and return a query error if the values are dangerous?

Reply:
The `placeholders` array is fine, I don't see where sanitation would happen with the `keys` string. The SQL library will definitely not be able to catch anything suspicious so it's up to us to sanitize. Lmk if this doesn't make sense.